pandas/core/tools/datetimes.py

```diff
@@ -79,16 +80,42 @@
 if TYPE_CHECKING:
     from pandas._libs.tslibs.nattype import NaTType
 
-    from pandas import Series
+    from pandas import (
+        DataFrame,
+        Series,
+    )
 
 # ---------------------------------------------------------------------
 # types used in annotations
 
-ArrayConvertible = Union[List, Tuple, AnyArrayLike, "Series"]
+ArrayConvertible = Union[List, Tuple, AnyArrayLike]
 Scalar = Union[int, float, str]
 DatetimeScalar = Union[Scalar, datetime]
 
 DatetimeScalarOrArrayConvertible = Union[DatetimeScalar, ArrayConvertible]
+
+DatetimeDictArg = Union[List[Scalar], Tuple[Scalar, ...], AnyArrayLike]
+
+
+class YearMonthDayDict(TypedDict, total=True):
+    year: DatetimeDictArg
+    month: DatetimeDictArg
+    day: DatetimeDictArg
+
+
+class FulldatetimeDict(YearMonthDayDict, total=False):
```

Comment:
II'm not sure what pandas's policy about moving types to `_typing.py` is. If you expect that this type will be useful in other places, it might be good to move it to `_typing.py`.

Reply:
I don't think it is used elsewhere, so will leave it here for now.